apps/website/src/pages/api/short-links/track-click.ts

```diff
@@ -0,0 +1,30 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { prisma } from "@/server/db/client";
+import { env } from "@/env/index.mjs";
+
+//API for short links tracking
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  const body = JSON.parse(req.body);
+  const id = body.id;
+  if (body.secret === env.SHORT_LINKS_TRACKING_SECRET) {
+    const query = await prisma.shortLinksTracking.findFirst({
+      where: { id: id },
+    });
```

Comment:
(Maybe look at the comment below regarding the creation of tracking entries first :) or this does not make as much sense)

Do we really need to query the entry first? If you use the upsert this should not be necessary. It would be possible that someone called the api with an invalid id, but that would require the correct api token so i do not think we really care. Even if this inserts a click for a non-existent link, it should not matter.